src/test/app/AMM_test.cpp

```diff
@@ -2986,6 +2985,208 @@ struct AMM_test : public jtx::AMMTest
         });
     }
 
+    // verifies: pool1Balance * pool2Balance >= lpTokenBalance^2
+    bool AMMInvariantCheck(STAmount pool1, STAmount pool2, STAmount lpToken) {
+        return pool1 * pool2 >= (lpToken * lpToken);
+    }
+
+    void
+    testExtremeValues() {
+        testcase("Extreme values test for AMM Pool Composition; Payments, "
+            "Deposits and Withdrawls against the AMM");
+
+        using namespace jtx;
+
+        // test the creation of AMM pools with a combination of the following
+        // initial XRP/USD balances
+        for (STAmount const& usd :
```

Comment:
@seelabs  is this what you meant by creating AMM pools with differing pool sizes? I'm not validating any other property of the AMMs except for their successful creation.